toxcore/group.c

```diff
@@ -3460,6 +3451,14 @@ static uint32_t load_group(Group_c *g, const Group_Chats *g_c, const uint8_t *da
             return 0;
         }
 
+        // This is inefficient, but allows us to check data consistency before allocating memory
+        g->frozen = (Group_Peer *)realloc(g->frozen, (j + 1) * sizeof(Group_Peer));
+
+        if (g->frozen == nullptr) {
+            // Memory allocation failure
```

Comment:
Memory leak: realloc doesn't free on failure.

Reply:
This is handled in the cleanup code in line 3572

Reply:
`g->frozen` will be `nullptr` and `free` will be a no-op. Where do you keep track of the old pointer that was passed to `realloc`?

Reply:
I see, should be fixed now.